src/trajectory_visualizer.cpp

```diff
@@ -44,51 +42,6 @@ void TrajectoryVisualizer::reset()
   points_ = std::make_unique<visualization_msgs::msg::MarkerArray>();
 }
 
-template <typename Container>
-void TrajectoryVisualizer::add(Container && trajectory)
```

Comment:
It doesn't appear to me that these need to be templated to have to be moved back to the header file. The 3 arg `add` and the 1 arg `add` are each only used in one place with `xt::xtensor<double, 2>` and `xt::xtensor<double, 3>` which are pretty standard across the repo. I think these could be un-templated and moved back to the header

Reply:
Sure, agree. I was going to do this after it starts to build and pass tests, so it would be easy to track if something could go wrong

Reply:
I just did it on my local branch as well, the next PR is going to have quite a few conflicts :laughing: we're both making big changes

Reply:
untemplated